### ISSUE TITLE
Datetime formatting when using milliseconds v2beta5

### ISSUE THREAD

**Issue by forestfairy** [constellation-app/constellation#620]
### Prerequisites

* [X ] Put an X between the brackets on this line if you have done all of the following:
    * Running the latest version of Constellation
	* Attached the ***Support Package*** via `Help` > `Support Package`
    * Checked the FAQs: https://github.com/constellation-app/constellation/wiki/FAQ
    * Checked that your issue isn't already filed: https://github.com/constellation-app/constellation/issues
    * Checked that there is not already a module that provides the described functionality: https://github.com/constellation-app/constellation/wiki/Catalogue-of-Repositories

### Description

Datetime formatting when using milliseconds

### Steps to Reproduce

1. Load Constellation
2. File>Import>
3. Allocate datetime attribute to parameters and format as a custom field

**Expected behaviour:** 

Expect most datetime formats to load,  

**Actual behaviour:** 

When using formatting that includes milliseconds, yyyy-MM-dd hh:mm:ss.xxxxxx (where x is the milliseconds) throws an error. Clean initial dataset to match dd/MM/yyyy HH:mm:ss zz format which does not throw an error in the import window, 

![image](https://user-images.githubusercontent.com/25699425/86429343-9b50ce00-bd32-11ea-8c1f-a77435eb7a8c.png)

This allows the import but errors out in the graph, it does not seem to be recognising the CUSTOM formatting for datetime ....

![image](https://user-images.githubusercontent.com/25699425/86429427-d0f5b700-bd32-11ea-8f11-98ea28543971.png)

**Reproduces how often:** 

100% at this time on the dataset with this format 

### Additional Information

Prevents the dataset from loading without some cleaning/loss of data

**Comment by serpens24** [comment]
The first part of the issue is that the supplied format is invalid, micro/milliseconds requires a .SSS/.SSSSSS extension rather than .xxx/.xxxxxx.

I tried using .SSSSSS and I no longer get an error, however the result was only resolved to 3 decimal points (milliseconds).

Three things need to be done:

1. Make it clear to users what formats are required to process milli/micro seconds, probably add an example/build in formatter for this.
2. Determine why only 3 decimal points are processed, this seems to be handled in utilties\temporal\TemporalFormatting.java which does not consider microseconds - refer to TIME_FORMATTER which references MILLI_OF_SECOND but not MICRO_OF_SECOND.

**Comment by serpens24** [comment]
The second part of the issue is to ensure the table within the import pane is able to accept values with milliseconds (s.SSS) and microseconds (s.SSSSSS) - currently only milliseconds is handled. 

In plugins\importexport\delimited\ImportTableColumn.validate() at line 86 takes the parsed value generated by the call on line 85 to DatetimeAttributeTranslator.translate (which in turn calls the TemporalFormatting code referred to above).
while the string returned at line 85 and stored in parsedValue looks correct for my test data - '2020-10-11 00:11.11.123456 +00:00', when this is passed to line 86 and error occurs, because the call to attributeDescription.acceptsString() makes a call to graph\attribute\ZonedDateTimeAttributeDescription.convertFromString which is expecting at most 3 values after the decimal point

**Comment by antares1470** [comment]
I've chatted with @forestfairy and they seem satisfied that switching to using S instead of x does the trick (as they were only after milliseconds rather than microseconds). If there is a desire to add support for microseconds, we can always create a new ticket or re-open this one. Therefore closing this ticket
